src/filter/http_context.rs

```diff
@@ -50,22 +54,31 @@ impl Filter {
             return Action::Continue;
         }
 
-        if let Some(operation) = self.operation_dispatcher.borrow_mut().next() {
-            match operation.get_result() {
+        match self.operation_dispatcher.borrow_mut().next() {
+            Ok(Some(op)) => match op.get_result() {
                 Ok(call_id) => {
                     debug!("#{} initiated gRPC call (id# {})", self.context_id, call_id);
                     Action::Pause
                 }
                 Err(e) => {
                     warn!("gRPC call failed! {e:?}");
-                    if let FailureMode::Deny = operation.get_failure_mode() {
+                    if let FailureMode::Deny = op.get_failure_mode() {
                         self.send_http_response(500, vec![], Some(b"Internal Server Error.\n"))
                     }
                     Action::Continue
                 }
+            },
+            Ok(None) => {
+                Action::Continue // No operations left to perform
             }
-        } else {
-            Action::Continue
+            Err(OperationError {
+                failure_mode: FailureMode::Deny,
+                ..
+            }) => {
+                self.send_http_response(500, vec![], Some(b"Internal Server Error.\n"));
+                Action::Continue
+            }
+            _ => Action::Continue,
```

Comment:
we could omit 

```rust
Ok(None) => {
                Action::Continue // No operations left to perform
            }
```

Left it there just so we now explicitly in case we want to perform something else

Reply:
So, for all _other_ `Err`, we ... `Action::Continue`?

Reply:
I guess I'm saying that I'd leave the `Ok(None)` => Action::Continue,` as an explicit branch _but_ would have 3 `Err` branches: 

 - `FailureMode::Deny`, as you do (log the actual error there? as this where it all ends, right?)
 - `FailureMode::Allow`, explicit with the actual error logged again
 - `Err(_)`, that possibly `panic!` or is marked as `unreachable!` as that I think is the assumption this builds upon

Reply:
I made it explicit the handling of both failure modes, but the third branch makes clippy unhappy. I'm also logging the status. Would that be enough?

Reply:
Let me know and I can include it in a following PR ;)